### PR TITLE
Feat/MET-5968 setting new data for target data

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

**Review by jeortizquan — CHANGES_REQUESTED** [review]
my suggestion to create a package and rename the dto's objects for the response, and also where it applies in other dto classes.

File: statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/TargetType.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    TOTAL_RECORDS("Total Records"),;
+
+    private final String valueAsString;
+
+    TargetType(String value){
+        this.valueAsString = value;
+    }
+
+    @JsonValue
+    public String getValueAsString() {
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/TargetType.java:18]: Changed it to `getName`

File: statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/HistoricalDataModel.java
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/HistoricalDataModel.java:1]: Done!

File: statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/TargetDataModel.java
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/internal/TargetDataModel.java:1]: Done!

File: statistics-dashboard-rest/src/main/java/eu/europeana/statistics/dashboard/rest/config/ApplicationConfiguration.java [hunk 2]
```diff
@@ -92,6 +93,12 @@ public StatisticsService getStatisticsService(MongoSDDao mongoSDDao){
     return statisticsService;
   }
 
+  @Bean
+  public TargetDataService getTargetDataService(MongoSDDao mongoSDDao){
+    return new TargetDataService(mongoSDDao);
+
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply statistics-dashboard-rest/src/main/java/eu/europeana/statistics/dashboard/rest/config/ApplicationConfiguration.java:99]: Removed!

File: statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/api/response/targetdata/CountryDataResult.java
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply statistics-dashboard-common/src/main/java/eu/europeana/statistics/dashboard/common/api/response/targetdata/CountryDataResult.java:1]: I changed it, I created a new package called `dto`, I changed the names to what I thought made sense according to your suggestions

**Review by jeortizquan — COMMENTED** [review]
There is the initialization of targets that can be improved, when the worker starts.

File: statistics-dashboard-worker/src/main/java/eu/europeana/statistics/dashboard/worker/execution/TargetDataRunner.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+//TODO: This script does not need to run frequently.
+//TODO: It should only run once to fill in with the latest data about target data
+//TODO: If that target data has changes, then this script needs to be updated
+public class TargetDataRunner implements CommandLineRunner {
+
+    private static final String COMMA_DELIMITER = ",";
+    private final ConfigurationPropertiesHolder propertiesHolder;
+    private static final Logger LOGGER = LoggerFactory.getLogger(AnalyzerRunner.class);
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply statistics-dashboard-worker/src/main/java/eu/europeana/statistics/dashboard/worker/execution/TargetDataRunner.java:28]: Fixed!

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**